lighthouse-core/computed/metrics/responsiveness.js

```diff
@@ -12,9 +12,34 @@
  */
 
 /** @typedef {LH.Trace.CompleteEvent & {name: 'Responsiveness.Renderer.UserInteraction', args: {frame: string, data: {interactionType: 'drag'|'keyboard'|'tapOrClick', maxDuration: number}}}} ResponsivenessEvent */
+/** @typedef {'keydown'|'keypress'|'keyup'|'mousedown'|'mouseup'|'pointerdown'|'pointerup'|'click'} EventTimingType */
+/**
+ * @typedef EventTimingData
+ * @property {string} frame
+ * @property {number} timeStamp The time of user interaction (in ms from navStart).
+ * @property {number} processingStart The start of interaction handling (in ms from navStart).
+ * @property {number} processingEnd The end of interaction handling (in ms from navStart).
+ * @property {number} duration The time from user interaction to browser paint (in ms).
+ * @property {EventTimingType} type
+ * @property {number} nodeId
+ * @property {number} interactionId
+ */
+/** @typedef {LH.Trace.AsyncEvent & {name: 'EventTiming', args: {data: EventTimingData}}} EventTimingEvent */
+
 
 const makeComputedArtifact = require('../computed-artifact.js');
 const ProcessedTrace = require('../processed-trace.js');
+const LHError = require('../../lib/lh-error.js');
+
+const KEYBOARD_EVENTS = new Set(['keydown', 'keypress', 'keyup']);
+const CLICK_TAP_DRAG_EVENTS = new Set([
+  'mousedown', 'mouseup', 'pointerdown', 'pointerup', 'click']);
+/** A map of Responsiveness `interactionType` to matching EventTiming `type`s. */
+const interactionTypeToType = {
+  keyboard: KEYBOARD_EVENTS,
+  tapOrClick: CLICK_TAP_DRAG_EVENTS,
+  drag: CLICK_TAP_DRAG_EVENTS,
+};
```

Comment:
Map?

Reply:
> Map?

It's probably better but in practice doesn't make any difference since it's not dynamic and the declaration is a lot uglier :)